### ISSUE TITLE
`Dynamic` type parameter casting issue

### ISSUE THREAD

**Issue by gene-pavlovsky** [HaxeFoundation/haxe#7246]
The issue is illustrated by this example:
https://try.haxe.org/#279ae
Note: `Future.NEVER` has type of `Future<Dynamic>`
The example uses tink_core's Futures but it doesn't have to...

A `Future<Dynamic>` can't be cast to `Future<Int>` when assigning to a variable, but it works using a trick with a type parameter function.

I think if the "trick" works, then a simple assignment should also work.
In other words, if it's ok to assign `Dynamic` to `Int`, why not `Future<Dynamic>` to `Future<Int>`
